pkg/operator/oauthapiencryptioncontroller/oauthapiencryptioncontroller.go

```diff
@@ -0,0 +1,128 @@
+	eventRecorder events.Recorder) factory.Controller {
+
+	controllerFactory := factory.New()
+	target := &oauthAPIServerController{
+		targetNamespace:               targetNamespace,
+		oauthAPIServerTargetNamespace: oauthAPIServerTargetNamespace,
+		secretLister:                  kubeInformersForNamespaces.InformersFor(operatorclient.GlobalMachineSpecifiedConfigNamespace).Core().V1().Secrets().Lister().Secrets(operatorclient.GlobalMachineSpecifiedConfigNamespace),
+		secretClient:                  client,
+		resourceSyncer:                resourceSyncer,
+	}
+
+	controllerFactory.WithSync(target.sync)
+	controllerFactory.WithInformers(kubeInformersForNamespaces.InformersFor(operatorclient.GlobalMachineSpecifiedConfigNamespace).Core().V1().Secrets().Informer())
+	return controllerFactory.ToController(name, eventRecorder.WithComponentSuffix("oauth-apiserver-encryption-cfg-sync-controller"))
+}
+
+// sync start managing oauth-apiserver encryption config (encryption-config-openshift-oauth-apiserver in openshift-config-managed namespace) as described in https://github.com/openshift/enhancements/blob/master/enhancements/etcd/etcd-encryption-for-separate-oauth-apis.md
+//
+// case 1: if the secret doesn't exist and encryption is on then:
+//         - it will add the secret with the annotation (must be atomic operation) because CAO will start managing its own config iff the secret exist without the annotation
+//         - it will start synchronisation
+//
+// case 2: if the secret exists and it is annotated
+//         - it will simply start synchronisation
+//
+// case 3: no-op: when the secret exits but it doesn't have the annotation - that means it was created by CAO in 4.6 and this is downgrade
+// case 4: no-op: when the secret doesn't exist and encryption is off
+//
+// drawbacks:
+// - it will not recover when the annotation was manually removed by a user,
+//   to recover we would have to put a value in the annotation instead and coordinate OAS-A and CAO but then we would have to remember to add it in CAO (4.6) as well
+func (c *oauthAPIServerController) sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	openshiftAPIServerEncryptionCfg, err := c.secretLister.Get(fmt.Sprintf("%s-%s", encryptionconfig.EncryptionConfSecretName, c.targetNamespace))
+	if apierrors.IsNotFound(err) {
+		return nil // case 4: encryption off
+	}
+	if err != nil {
+		return err
+	}
+
+	oauthAPIServerEncryptionCfg, err := c.secretLister.Get(fmt.Sprintf("%s-%s", encryptionconfig.EncryptionConfSecretName, c.oauthAPIServerTargetNamespace))
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	if apierrors.IsNotFound(err) {
+		// case 1: create with annotation and start synchronization
+		source := openshiftAPIServerEncryptionCfg.DeepCopy()
+		source.Name = fmt.Sprintf("%s-%s", encryptionconfig.EncryptionConfSecretName, c.oauthAPIServerTargetNamespace)
+		source.ResourceVersion = ""
+		source.OwnerReferences = []metav1.OwnerReference{}
```

Comment:
would rather start with a fresh ObjectMeta and copy everything over you want. There are other fields we probably don't want here.

Reply:
finalizers, managedFields, customer annotations, labels, ...

Reply:
since I'm relying on `c.resourceSyncer.SyncSecret` annotations and labels will be copied anyway but I get rid of finalizers.